casanovo/denovo/model.py

```diff
@@ -1059,14 +1059,16 @@ def predict_step(self, batch, *args):
             # Use _forward_step
```

Comment:
I would factorize the batch construction here into a separate method and add a quick unit test for it - looks correct now, but we want to make sure it keeps working if/when we change file formats. Weird predictions stemming from peptides being matched to the wrong spectra/precursors would be pretty annoying to track down.